assembl/graphql/schema.py

```diff
@@ -1,81 +1,439 @@
+        query = get_query(model, context)
+        discussion_id = context.matchdict['discussion_id']
+        query = query.filter(model.discussion_id == discussion_id
+            ).filter(model.hidden == False
+            ).filter(model.tombstone_condition())
+
+        # pagination is done after that, no need to do it ourself
+        return query
+
+    def resolve_posts(self, args, context, info):
+        connection_type = info.return_type.graphene_type  # this is PostConnection
+        model = connection_type._meta.node._meta.model  # this is models.PostUnion
+        query = get_query(model, context)
+        discussion_id = context.matchdict['discussion_id']
+        query = query.filter(model.discussion_id == discussion_id
+            ).filter(model.hidden == False
+            ).filter(model.tombstone_condition()
+            ).filter(model.publication_state == models.PublicationStates.PUBLISHED)
+
+        # TODO filter posts for a specific idea
+        # idea_id = args.get('idea_id', None)
+        # if idea_id is not None:
+        #     id_ = int(Node.from_global_id(idea_id)[1])
+        #     query = query.filter(model.idea_content_links? idea_id? == id_)
+
+        # pagination is done after that, no need to do it ourself
+        return query
+
+    def resolve_thematics(self, args, context, info):
+        identifier = args.get('identifier', None)
+        model = Thematic._meta.model
+        discussion_id = context.matchdict['discussion_id']
+        query = get_query(model, context
+            ).filter(model.discussion_id == discussion_id)
+        if identifier is not None:
+            query = query.filter(model.identifier == identifier)
+
+        return query
+
+
+class LangStringEntryFields(graphene.AbstractType):
+    value = graphene.String(required=True)
+    locale_code = graphene.String(required=True)
+
+
+class LangStringEntry(graphene.ObjectType, LangStringEntryFields):
+    pass
+
+
+class LangStringEntryInput(graphene.InputObjectType, LangStringEntryFields):
+    pass
+
+
+class VideoInput(graphene.InputObjectType):
+    title_entries = graphene.List(LangStringEntryInput)
+    description_entries = graphene.List(LangStringEntryInput)
+    html_code = graphene.String()
+
+
+class QuestionInput(graphene.InputObjectType):
+    title_entries = graphene.List(LangStringEntryInput, required=True)
+
+
+class CreateThematic(graphene.Mutation):
+    class Input:
+        # Careful, having required=True on a graphene.List only means
+        # it can't be None, having an empty [] is perfectly valid.
+        title_entries = graphene.List(LangStringEntryInput, required=True)
+        description_entries = graphene.List(LangStringEntryInput)
+        identifier = graphene.String(required=True)
+        video = graphene.Argument(VideoInput)
+        questions = graphene.List(QuestionInput)
+        # TODO upload img example http://docs.pylonsproject.org/projects/pyramid-cookbook/en/latest/forms/file_uploads.html
```

Comment:
This is what you want to look at: https://github.com/homeslicesolutions/backbone-model-file-upload